nautobot/extras/tests/test_views.py

```diff
@@ -988,28 +1036,234 @@ def setUpTestData(cls):
+
+        response_body = extract_page_body(response.content.decode(response.charset))
+        self.assertIn("TestPass", response_body)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=[], JOBS_ROOT=DUMMY_JOBS)
+    def test_post_without_permission(self):
+        response = self.client.post(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}))
+        self.assertHttpStatus(response, 403)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    def test_run_without_schedule(self):
+        self.add_permissions("extras.run_job")
+
+        response = self.client.post(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}))
+
+        self.assertHttpStatus(response, 200)
+        errors = extract_form_failures(response.content.decode(response.charset))
+        self.assertEqual(errors, ["_schedule_type: This field is required."])
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    def test_run_now_no_worker(self):
+        self.add_permissions("extras.run_job")
+
+        data = {
+            "_schedule_type": "immediately",
+        }
+
+        response = self.client.post(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}), data)
+
+        self.assertHttpStatus(response, 200)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    @mock.patch("nautobot.extras.utils.get_worker_count")
+    def test_run_now(self, patched):
+        self.add_permissions("extras.run_job")
+
+        data = {
+            "_schedule_type": "immediately",
+        }
+
+        patched.return_value = 1
+        response = self.client.post(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}), data)
+
+        self.assertHttpStatus(response, 302)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    def test_run_now_missing_args(self):
+        self.add_permissions("extras.run_job")
+
+        data = {
+            "_schedule_type": "immediately",
+        }
+
+        response = self.client.post(
+            reverse("extras:job", kwargs={"class_path": "local/test_required_args/TestRequired"}), data
+        )
+
+        self.assertHttpStatus(response, 200)
+        errors = extract_form_failures(response.content.decode(response.charset))
+        self.assertEqual(errors, ["var: This field is required."])
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    @mock.patch("nautobot.extras.utils.get_worker_count")
+    def test_run_now_with_args(self, patched):
+        self.add_permissions("extras.run_job")
+
+        data = {
+            "_schedule_type": "immediately",
+            "var": "12",
+        }
+
+        patched.return_value = 1
+        response = self.client.post(
+            reverse("extras:job", kwargs={"class_path": "local/test_required_args/TestRequired"}), data
+        )
+
+        self.assertHttpStatus(response, 302)
```

Comment:
Same comments here about assertRedirects and checking that enqueue_job was called correctly.

Reply:
I’ve changed it to `assertRedirect` here as well.